specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,353 @@
+    const videoTrack = video_tracks[0];
+    // Show the video via Video Element.
+    document.getElementById(video_id).srcObject = stream;
+  } catch (error) {
+    console.log(error);
+  }
+}
+Win32 C++
+UINT32 luid,
+// Get the LUID (Graphic adapter) that the WebView renderer uses.
+coreWebView->GetRenderAdapterLUID(&luid);
+// Create D3D device based on the WebView's LUID.
+ComPtr<D3D11Device> d3d_device = MyCreateD3DDevice(luid);
+// Register unique texture stream that the host can provide. 
+ComPtr<ICoreWebView2TextureStream> webviewTextureStream;
+g_webviewStaging3->CreateTextureStream(L"webview2-abcd1234", d3d_device.Get(),  &webviewTextureStream);
+// Register the Origin URL that the target renderer could stream of the registered stream id. The request from not registered origin will fail to stream.
+webviewTextureStream->AddRequestedFilter(L"https://edge-webscratch");
+// Listen to Start request
+EventRegistrationToken start_token;
+webviewTextureStream->add_StartRequested(Callback<ICoreWebView2StagingTextureStreamStartRequestedEventHandler>(
+  [hWnd](ICoreWebView2StagingTextureStream* webview, IUnknown* eventArgs) -> HRESULT {
+    // Capture video stream by using native API, for example, Media Foundation on Windows.
+    StartMediaFoundationCapture(hWnd);
+    return S_OK;
+  }).Get(), &start_token);
+EventRegistrationToken stop_token;
+webviewTextureStream->add_StopRequested(Callback<ICoreWebView2StagingTextureStreamStopRequestedEventHandler>(
+  [hWnd](ICoreWebView2StagingTextureStream* webview, IUnknown* eventArgs) -> HRESULT {
+    StopMediaFoundationCapture();
+    return S_OK;
+  }).Get(), &stop_token);
+EventRegistrationToken texture_token;
+webviewTextureStream->add_TextureError(Callback<ICoreWebView2StagingTextureStreamTextureErrorEventHandler>(
+  [hWnd](ICoreWebView2StagingTextureStream* sender, ICoreWebView2StagingTextureStreamTextureErrorEventArgs* args) {
+    COREWEBVIEW2_TEXTURE_STREAM_ERROR_KIND kind;
+    HRESULT hr = args->get_Kind(&kind);
+    assert(SUCCEEDED(hr));
+    switch (kind)
+    {
+    case COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED:
+    case COREWEBVIEW2_TEXTURE_STREAM_ERROR_BUFFER_NOT_FOUND:
+    case COREWEBVIEW2_TEXTURE_STREAM_ERROR_BUFFER_IN_USE:
+      // assert(false);
+      break;
+    default:
+      break;
+    }
+    return S_OK;
+  }).Get(), &texture_token);
+    
+// TextureStream APIs are called in the UI thread on the WebView2 process meanwhile Video capture
+// and composition could happen in worker thread or out of process.
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+  static ICoreWebView2Staging3* webview2_17 = nullptr;
+  TCHAR greeting[] = _T("Hello, Windows desktop!");
+  ComPtr<ID3D11Device> d3d_device;
+  HANDLE slimCoreHandle;
+  HRESULT hr;
+  ComPtr<ICoreWebView2StagingTexture> texture_buffer;
+  int64_t bufferId = -1;
+  switch (message)
+  {
+  case IDC_TEST_SEND_TEXTURE:
```

Comment:
Where do these window messages come from? If we don't perform this part of the sample in a different process, would it make the sample smaller and easier to understand?

Reply:
I've updated it to be simpler, but let me know if you think it needs more changes.